// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/CreateChannelMembershipCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateChannelMembershipRequestFilterSensitiveLog, CreateChannelMembershipResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateChannelMembershipCommand, serializeAws_restJson1CreateChannelMembershipCommand, } from "../protocols/Aws_restJson1";
var CreateChannelMembershipCommand = (function (_super) {
    __extends(CreateChannelMembershipCommand, _super);
    function CreateChannelMembershipCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateChannelMembershipCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "CreateChannelMembershipCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateChannelMembershipRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateChannelMembershipResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateChannelMembershipCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateChannelMembershipCommand(input, context);
    };
    CreateChannelMembershipCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateChannelMembershipCommand(output, context);
    };
    return CreateChannelMembershipCommand;
}($Command));
export { CreateChannelMembershipCommand };
