// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/DeleteChannelMessageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteChannelMessageRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteChannelMessageCommand, serializeAws_restJson1DeleteChannelMessageCommand, } from "../protocols/Aws_restJson1";
var DeleteChannelMessageCommand = (function (_super) {
    __extends(DeleteChannelMessageCommand, _super);
    function DeleteChannelMessageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteChannelMessageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "DeleteChannelMessageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteChannelMessageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteChannelMessageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteChannelMessageCommand(input, context);
    };
    DeleteChannelMessageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteChannelMessageCommand(output, context);
    };
    return DeleteChannelMessageCommand;
}($Command));
export { DeleteChannelMessageCommand };
