// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/DescribeChannelMembershipCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeChannelMembershipRequestFilterSensitiveLog, DescribeChannelMembershipResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1DescribeChannelMembershipCommand, serializeAws_restJson1DescribeChannelMembershipCommand, } from "../protocols/Aws_restJson1";
var DescribeChannelMembershipCommand = (function (_super) {
    __extends(DescribeChannelMembershipCommand, _super);
    function DescribeChannelMembershipCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeChannelMembershipCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "DescribeChannelMembershipCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeChannelMembershipRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeChannelMembershipResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeChannelMembershipCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeChannelMembershipCommand(input, context);
    };
    DescribeChannelMembershipCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeChannelMembershipCommand(output, context);
    };
    return DescribeChannelMembershipCommand;
}($Command));
export { DescribeChannelMembershipCommand };
