// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/DescribeChannelFlowCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeChannelFlowRequestFilterSensitiveLog, DescribeChannelFlowResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1DescribeChannelFlowCommand, serializeAws_restJson1DescribeChannelFlowCommand, } from "../protocols/Aws_restJson1";
var DescribeChannelFlowCommand = (function (_super) {
    __extends(DescribeChannelFlowCommand, _super);
    function DescribeChannelFlowCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeChannelFlowCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "DescribeChannelFlowCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeChannelFlowRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeChannelFlowResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeChannelFlowCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeChannelFlowCommand(input, context);
    };
    DescribeChannelFlowCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeChannelFlowCommand(output, context);
    };
    return DescribeChannelFlowCommand;
}($Command));
export { DescribeChannelFlowCommand };
