// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/DescribeChannelModeratorCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeChannelModeratorRequestFilterSensitiveLog, DescribeChannelModeratorResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1DescribeChannelModeratorCommand, serializeAws_restJson1DescribeChannelModeratorCommand, } from "../protocols/Aws_restJson1";
var DescribeChannelModeratorCommand = (function (_super) {
    __extends(DescribeChannelModeratorCommand, _super);
    function DescribeChannelModeratorCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeChannelModeratorCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "DescribeChannelModeratorCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeChannelModeratorRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeChannelModeratorResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeChannelModeratorCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeChannelModeratorCommand(input, context);
    };
    DescribeChannelModeratorCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeChannelModeratorCommand(output, context);
    };
    return DescribeChannelModeratorCommand;
}($Command));
export { DescribeChannelModeratorCommand };
