// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/GetChannelMembershipPreferencesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetChannelMembershipPreferencesRequestFilterSensitiveLog, GetChannelMembershipPreferencesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetChannelMembershipPreferencesCommand, serializeAws_restJson1GetChannelMembershipPreferencesCommand, } from "../protocols/Aws_restJson1";
var GetChannelMembershipPreferencesCommand = (function (_super) {
    __extends(GetChannelMembershipPreferencesCommand, _super);
    function GetChannelMembershipPreferencesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetChannelMembershipPreferencesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "GetChannelMembershipPreferencesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetChannelMembershipPreferencesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetChannelMembershipPreferencesResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetChannelMembershipPreferencesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetChannelMembershipPreferencesCommand(input, context);
    };
    GetChannelMembershipPreferencesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetChannelMembershipPreferencesCommand(output, context);
    };
    return GetChannelMembershipPreferencesCommand;
}($Command));
export { GetChannelMembershipPreferencesCommand };
