// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/ListChannelFlowsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListChannelFlowsRequestFilterSensitiveLog, ListChannelFlowsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1ListChannelFlowsCommand, serializeAws_restJson1ListChannelFlowsCommand, } from "../protocols/Aws_restJson1";
var ListChannelFlowsCommand = (function (_super) {
    __extends(ListChannelFlowsCommand, _super);
    function ListChannelFlowsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListChannelFlowsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "ListChannelFlowsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListChannelFlowsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListChannelFlowsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListChannelFlowsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListChannelFlowsCommand(input, context);
    };
    ListChannelFlowsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListChannelFlowsCommand(output, context);
    };
    return ListChannelFlowsCommand;
}($Command));
export { ListChannelFlowsCommand };
