// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/CreateChannelBanCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateChannelBanRequestFilterSensitiveLog, CreateChannelBanResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateChannelBanCommand, serializeAws_restJson1CreateChannelBanCommand, } from "../protocols/Aws_restJson1";
var CreateChannelBanCommand = (function (_super) {
    __extends(CreateChannelBanCommand, _super);
    function CreateChannelBanCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateChannelBanCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "CreateChannelBanCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateChannelBanRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateChannelBanResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateChannelBanCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateChannelBanCommand(input, context);
    };
    CreateChannelBanCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateChannelBanCommand(output, context);
    };
    return CreateChannelBanCommand;
}($Command));
export { CreateChannelBanCommand };
