// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/DisassociateChannelFlowCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisassociateChannelFlowRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DisassociateChannelFlowCommand, serializeAws_restJson1DisassociateChannelFlowCommand, } from "../protocols/Aws_restJson1";
var DisassociateChannelFlowCommand = (function (_super) {
    __extends(DisassociateChannelFlowCommand, _super);
    function DisassociateChannelFlowCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateChannelFlowCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "DisassociateChannelFlowCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateChannelFlowRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateChannelFlowCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisassociateChannelFlowCommand(input, context);
    };
    DisassociateChannelFlowCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisassociateChannelFlowCommand(output, context);
    };
    return DisassociateChannelFlowCommand;
}($Command));
export { DisassociateChannelFlowCommand };
