// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/ListChannelsModeratedByAppInstanceUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListChannelsModeratedByAppInstanceUserRequestFilterSensitiveLog, ListChannelsModeratedByAppInstanceUserResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1ListChannelsModeratedByAppInstanceUserCommand, serializeAws_restJson1ListChannelsModeratedByAppInstanceUserCommand, } from "../protocols/Aws_restJson1";
var ListChannelsModeratedByAppInstanceUserCommand = (function (_super) {
    __extends(ListChannelsModeratedByAppInstanceUserCommand, _super);
    function ListChannelsModeratedByAppInstanceUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListChannelsModeratedByAppInstanceUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "ListChannelsModeratedByAppInstanceUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListChannelsModeratedByAppInstanceUserRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListChannelsModeratedByAppInstanceUserResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListChannelsModeratedByAppInstanceUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListChannelsModeratedByAppInstanceUserCommand(input, context);
    };
    ListChannelsModeratedByAppInstanceUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListChannelsModeratedByAppInstanceUserCommand(output, context);
    };
    return ListChannelsModeratedByAppInstanceUserCommand;
}($Command));
export { ListChannelsModeratedByAppInstanceUserCommand };
