// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/ListChannelMembershipsForAppInstanceUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListChannelMembershipsForAppInstanceUserRequestFilterSensitiveLog, ListChannelMembershipsForAppInstanceUserResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1ListChannelMembershipsForAppInstanceUserCommand, serializeAws_restJson1ListChannelMembershipsForAppInstanceUserCommand, } from "../protocols/Aws_restJson1";
var ListChannelMembershipsForAppInstanceUserCommand = (function (_super) {
    __extends(ListChannelMembershipsForAppInstanceUserCommand, _super);
    function ListChannelMembershipsForAppInstanceUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListChannelMembershipsForAppInstanceUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "ListChannelMembershipsForAppInstanceUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListChannelMembershipsForAppInstanceUserRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListChannelMembershipsForAppInstanceUserResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListChannelMembershipsForAppInstanceUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListChannelMembershipsForAppInstanceUserCommand(input, context);
    };
    ListChannelMembershipsForAppInstanceUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListChannelMembershipsForAppInstanceUserCommand(output, context);
    };
    return ListChannelMembershipsForAppInstanceUserCommand;
}($Command));
export { ListChannelMembershipsForAppInstanceUserCommand };
