// @aws-sdk/client-chime-sdk-messaging/dist-es/commands/UpdateChannelMessageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateChannelMessageRequestFilterSensitiveLog, UpdateChannelMessageResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateChannelMessageCommand, serializeAws_restJson1UpdateChannelMessageCommand, } from "../protocols/Aws_restJson1";
var UpdateChannelMessageCommand = (function (_super) {
    __extends(UpdateChannelMessageCommand, _super);
    function UpdateChannelMessageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateChannelMessageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ChimeSDKMessagingClient";
        var commandName = "UpdateChannelMessageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateChannelMessageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateChannelMessageResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateChannelMessageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateChannelMessageCommand(input, context);
    };
    UpdateChannelMessageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateChannelMessageCommand(output, context);
    };
    return UpdateChannelMessageCommand;
}($Command));
export { UpdateChannelMessageCommand };
